hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java

```diff
@@ -699,9 +693,9 @@ private static String escapeSpecialValueIfNeeded(String value) {
         if (value.isEmpty()) {
             // Escape empty String to INDEX_SYM_EMPTY (char `\u0002`)
             value = INDEX_SYM_EMPTY;
-        } else if (value == INDEX_VALUE_EMPTY) {
+        } else if (INDEX_VALUE_EMPTY.equals(value)) {
```

Comment:
must keep "==" here since INDEX_VALUE_EMPTY is just a special symbol, we can add some comments for it